### PR TITLE
[BTFSINFRA-467] return error code on startup test failure

### PR DESCRIPTION
**Pull request by hzmbeyond80**
retry the start up test twice and return error code if fails.

### CODE REVIEW

File: autoupdate/main.go [hunk 1]
```diff
@@ -241,15 +241,32 @@ func update(log *zap.Logger) int {
...
+			} else {
+				log.Error("BTFS daemon get file test failed!")
+			}
 		}
-		if add_functest(log, btfsBinaryPath) {
-			log.Info("BTFS daemon add file test succeeded!")
-		} else {
-			log.Error("BTFS daemon add file test failed!")
+		if !test_success {
+			os.Exit(101)
```
Comment (cong201808) [inline autoupdate/main.go]: Can we define the error code instead of being a random number here?

File: autoupdate/main.go [hunk 2]
```diff
@@ -241,15 +241,32 @@ func update(log *zap.Logger) int {
...
+		for i := 0; i < 2; i++ {
+			if add_functest(log, btfsBinaryPath) {
+				log.Info("BTFS daemon add file test succeeded!")
+				test_success = true
+			} else {
+				log.Error("BTFS daemon add file test failed!")
+			}
+		}
+		if !test_success {
+			os.Exit(102)
```
Comment (cong201808) [inline autoupdate/main.go]: Ditto

**Comment by richlee33** [comment]
Zhimin and i tested this by making a new branch on btfs-binary-releases https://github.com/TRON-US/btfs-binary-releases/tree/test2 which has a later version 0.1.1 and changing the autoupdate URL to point to that branch.  the new autoupdate bin ran fine.
that being said, adding new unit tests would be a good idea so i will add a new story for this.

**Comment by cong201808** [comment]
Thanks. To be clear, unit tests are responsibilities of those who write the code, not SET's.

File: autoupdate/main.go (hunk 1 above)
Comment (hzmbeyond80) [reply autoupdate/main.go]: i had felt it's optional to name a variable that's only used once.  but i agree it helps readability.  Jin is making a change on this file in his PR and i have requested him to put in the change while he is there.

**Pull request merged**